x-pack/plugin/mapper-patterned-text/src/main/java/org/elasticsearch/xpack/patternedtext/PatternedTextFieldMapper.java

```diff
@@ -0,0 +1,168 @@
+            return new PatternedTextFieldType(
+                context.buildFullName(leafName()),
+                tsi,
+                indexAnalyzer,
+                context.isSourceSynthetic(),
+                meta.getValue()
+            );
+        }
+
+        @Override
+        public PatternedTextFieldMapper build(MapperBuilderContext context) {
+            return new PatternedTextFieldMapper(leafName(), buildFieldType(context), builderParams(this, context), this);
+        }
+    }
+
+    public static final TypeParser PARSER = new TypeParser((n, c) -> new Builder(n, c.indexVersionCreated(), c.getIndexAnalyzers()));
+
+    private final IndexVersion indexCreatedVersion;
+    private final IndexAnalyzers indexAnalyzers;
+    private final NamedAnalyzer indexAnalyzer;
+    private final int positionIncrementGap;
+    private final FieldType fieldType;
+
+    private PatternedTextFieldMapper(
+        String simpleName,
+        PatternedTextFieldType mappedFieldPatternedTextFieldType,
+        BuilderParams builderParams,
+        Builder builder
+    ) {
+        super(simpleName, mappedFieldPatternedTextFieldType, builderParams);
+        assert mappedFieldPatternedTextFieldType.getTextSearchInfo().isTokenized();
+        assert mappedFieldPatternedTextFieldType.hasDocValues();
+        this.fieldType = Defaults.FIELD_TYPE;
+        this.indexCreatedVersion = builder.indexCreatedVersion;
+        this.indexAnalyzers = builder.analyzers.indexAnalyzers;
+        this.indexAnalyzer = builder.analyzers.getIndexAnalyzer();
+        this.positionIncrementGap = builder.analyzers.positionIncrementGap.getValue();
+    }
+
+    @Override
+    public Map<String, NamedAnalyzer> indexAnalyzers() {
+        return Map.of(mappedFieldType.name(), indexAnalyzer);
+    }
+
+    @Override
+    public FieldMapper.Builder getMergeBuilder() {
+        return new Builder(leafName(), indexCreatedVersion, indexAnalyzers).init(this);
+    }
+
+    @Override
+    protected void parseCreateField(DocumentParserContext context) throws IOException {
+        final String value = context.parser().textOrNull();
+        if (value == null) {
+            return;
+        }
+
+        // Parse template and args.
+        PatternedTextValueProcessor.Parts parts = PatternedTextValueProcessor.split(value);
+
+        // Add index on original value
+        context.doc().add(new Field(fieldType().name(), value, fieldType));
+
+        // Add template doc_values
+        context.doc().add(new SortedSetDocValuesField(fieldType().templateFieldName(), new BytesRef(parts.template())));
+
+        // Add args doc_values
+        if (parts.args().isEmpty() == false) {
+            String remainingArgs = PatternedTextValueProcessor.encodeRemainingArgs(parts);
+            context.doc().add(new SortedSetDocValuesField(fieldType().argsFieldName(), new BytesRef(remainingArgs)));
```

Comment:
Like an earlier comment, both `SortedSetDocValuesField` usages can be replaced here with `SortedDocValuesField `